### ISSUE TITLE
Create and Submit Architecture Documentation

### ISSUE THREAD

**Issue by pachonk** [emerald-rubies/greenhouse#3]

**Comment by pachonk** [comment]
I think after the initial project chores are complete, we should have some people hop on a video call to get this done together.

**Comment by pachonk** [comment]
We could just start commenting the user stories here for now.
@dendenvitorio can you give a detailed description of what you want the project to do before we release to the general public, then we can split it into "user stories"?

**Comment by dendenvitorio** [comment]
Hi @pachonk ! Sure! 

As a Freelancer, I want to be able to Sign Up so that I can register and potential employers may see my profile.

As an Employer, I want to be able to Sign Up so that I can see the list of potential Freelancers I can hire

**Comment by pachonk** [comment]
I don't _love_ the DSL "freelancer". Maybe we could use "developer" and "client"? Not all people buying development work are considered "employers" to me.

**Comment by pachonk** [comment]
Ehh, client is too general too.

**Comment by dendenvitorio** [comment]
Users... ?

**Comment by pachonk** [comment]
`Users` would be the model that they have, but we still need to have a front end "name" for the 2 groups of people.
I think maybe "organization" and "developer"?

**Comment by pachonk** [comment]
OHHH WAIT, do you want to make this literally a freelancing site?

**Comment by dendenvitorio** [comment]
So basically, my first thought or idea with this one is for people who are struggling to find some helpers that will either "Stay-In" or "Stay-Out". Then employers will find helpers with good reputation in this review site that they want to hire.

**Comment by pachonk** [comment]
@dendenvitorio I'm perfectly fine if this is simply for experience, but you are aware that MANY sites do this exact thing, so it will have a rather low adoption rate. If you're okay with that, we can proceed. If you're looking for an app that people will use/enjoy, then maybe we should open a brainstorming session.
I misunderstood the original intention of this app.

**Comment by dendenvitorio** [comment]
Okay let's proceed to freelancing site. I'm okay with that too.

**Comment by pachonk** [comment]
Your most recent comment confuses me. So we're going to proceed with this mini project, correct?

**Comment by dendenvitorio** [comment]
Oh sorry, forget my comment:

So basically, my first thought or idea with this one is for people who are struggling to find some helpers that will either "Stay-In" or "Stay-Out". Then employers will find helpers with good reputation in this review site that they want to hire.

Let's proceed with your comment:

Users would be the model that they have, but we still need to have a front end "name" for the 2 groups of people.
I think maybe "organization" and "developer"?

I am fine with Organization and Developer.

**Comment by pachonk** [comment]
So basically this site is a search engine for employers to search for developers with metrics based on previous ratings and open a line of communication between the two?

**Comment by dendenvitorio** [comment]
Correct. Also, employers can also do job postings and developers will apply and employers will review them and such. You getting the idea or not?

**Comment by pachonk** [comment]
I get the idea.
Check this site out: http://www.greenhouse.io/
They do the almost the same thing with the same name. Since this is a trial project, I don't feel the need to change it, but I did want to point that out

**Comment by pachonk** [comment]
Do developers need to sign up before they can be rated? Or could employers make a profile for a dev and the dev can later claim it?

**Comment by Exellin** [comment]
What exactly do "helpers" and "stay-in" or "stay-out" mean for this project?

**Comment by dendenvitorio** [comment]
Hey Shawn, ignore that Helpers feature. 

Alex, Developers need to sign up.

**Comment by pachonk** [comment]
How do you feel about these so far?
- [ ] As a developer, I want to sign up and create a resume/profile
- [ ] As an employer, I want to be able to sign up and review developers
- [ ] As an employer, I want to be able to sign up and search for developers based on a specified set of metrics (4-5 star rating on Ruby, 85% minimum customer satisfaction, etc)
